EDDiscovery/Program.cs

```diff
@@ -31,15 +63,40 @@ static void Main()
             }
             catch (TimeoutException)
             {
-                if (MessageBox.Show("EDDiscovery is already running. Launch anyway?", "EDDiscovery", MessageBoxButtons.YesNo) == DialogResult.Yes)
+                /* Could not lock the app-global mutex, which means another copy of the App is running.
+                 * Let's try and show it. If we can't, then fallback to showing the classic MessageBox.
+                 */
+                if (!RaiseOtherProcess() && MessageBox.Show("EDDiscovery is already running. Launch anyway?", "EDDiscovery", MessageBoxButtons.YesNo) == DialogResult.Yes)
                 {
                     Application.Run(new EDDiscoveryForm());
                 }
+            }
+        }
 
-                /* Could not lock the app-global mutex, which means another copy of the App is running.
-                 * TODO: show a dialog and/or bring the current instance's window to the foreground.
-                 */
+        /// <summary>
+        /// Attempt to bring the main window of a different process to the foreground.
+        /// This is really only useful if we are being launched twice because the user
+        /// was not aware that the application was already running.
+        /// </summary>
+        /// <returns><c>true</c> if the other application windw was raised. <c>false</c> otherwise</returns>
+        static bool RaiseOtherProcess()
+        {
+            bool retval = false;
+            Process proc = Process.GetCurrentProcess();
+            foreach (Process otherProc in Process.GetProcessesByName(proc.ProcessName))
+            {
+                if (proc.Id != otherProc.Id)
+                {
+                    IntPtr hWnd = otherProc.MainWindowHandle;
```

Comment:
A portable alternative might be:
```
Microsoft.VisualBasic.Interaction.AppActivate(otherProc.Id);
```
Though that won't restore the app if it's minimized.

This block could be wrapped in `try { } catch (DllNotFoundException) { } catch { }`, where the portable function is called if binding the Win32 functions fail, and `retval` is set to `false` if any other exception is caught.